internal/services/signalr/web_pubsub_custom_certificate_binding.go

```diff
@@ -0,0 +1,282 @@
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.Any(
+				keyVaultValidate.NestedItemId,
+				keyVaultValidate.NestedItemIdWithOptionalVersion,
+			),
+		},
+
+		"certificate_version": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+	}
+}
+
+func (r CustomCertBindingWebPubsubResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r CustomCertBindingWebPubsubResource) ModelObject() interface{} {
+	return &CustomCertBindingWebPubsubModel{}
+}
+
+func (r CustomCertBindingWebPubsubResource) ResourceType() string {
+	return "azurerm_web_pubsub_custom_certificate_binding"
+}
+
+func (r CustomCertBindingWebPubsubResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var webPubsubCustomCertBinding CustomCertBindingWebPubsubModel
+			if err := metadata.Decode(&webPubsubCustomCertBinding); err != nil {
+				return err
+			}
+			client := metadata.Client.SignalR.WebPubSubClient.WebPubSub
+
+			webPubsubId, err := webpubsub.ParseWebPubSubID(metadata.ResourceData.Get("web_pubsub_id").(string))
+			if err != nil {
+				return fmt.Errorf("parsing web pubsub service id error: %+v", err)
+			}
+
+			keyVaultCertificateId, err := keyVaultParse.ParseOptionallyVersionedNestedItemID(metadata.ResourceData.Get("custom_certificate_id").(string))
+			if err != nil {
+				return fmt.Errorf("parsing custom certificate id error: %+v", err)
+			}
+
+			keyVaultUri := keyVaultCertificateId.KeyVaultBaseUrl
+			keyVaultSecretName := keyVaultCertificateId.Name
+
+			id := webpubsub.NewCustomCertificateID(webPubsubId.SubscriptionId, webPubsubId.ResourceGroupName, webPubsubId.WebPubSubName, metadata.ResourceData.Get("name").(string))
+
+			existing, err := client.CustomCertificatesGet(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing web pubsub service custom cert binding error %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			customCert := webpubsub.CustomCertificate{
+				Properties: webpubsub.CustomCertificateProperties{
+					KeyVaultBaseUri:    keyVaultUri,
+					KeyVaultSecretName: keyVaultSecretName,
+				},
+			}
+
+			if certVersion := keyVaultCertificateId.Version; certVersion != "" {
+				if webPubsubCustomCertBinding.CertificateVersion != "" && certVersion != webPubsubCustomCertBinding.CertificateVersion {
+					return fmt.Errorf("certificate version in cert id is different from `certificate_version`")
+				}
+				customCert.Properties.KeyVaultSecretVersion = utils.String(certVersion)
+			}
+
+			if _, err := client.CustomCertificatesCreateOrUpdate(ctx, id, customCert); err != nil {
```

Comment:
Would `CustomCertificatesCreateOrUpdateThenPoll` work here instead of using `WaitForStateContext`?
```suggestion
			if _, err := client.CustomCertificatesCreateOrUpdate(ctx, id, customCert); err != nil {
```